### PR TITLE
Bug fix with concat of empty lists.

### PR DESCRIPTION
**Pull request by bmcgary**
Oops.😳  Forgot about the edge cases in my first PR (#3 ), sorry about that. This should take care of that.

### CODE REVIEW

File: test/list.spec.js [hunk 1]
```diff
@@ -265,6 +265,27 @@ describe('Linked List', function() {
             arrayedLists.forEach(function(d, i) {
                 expect(d).to.equal(newList.findAt(i).getData());
             });
+
+            //this list is empty, second is not
+            list.clear();
+            newList = list.concat(secondList);
+            expect(newList.getSize(), 5);
+            expect(newList.getHeadNode()).to.not.equal(null);
+            expect(newList.getTailNode()).to.not.equal(null);
```
Comment (jasonsjones) [inline test/list.spec.js:276]: We have a lot expectations in this test...and we are testing different cases, so maybe we should split these out to their own tests.  Maybe something like...

```
it('when empty concats a populated list', () => {
    // set up and expectations here
});
```

File: test/list.spec.js [hunk 2]
```diff
@@ -265,6 +265,27 @@ describe('Linked List', function() {
...
+
+            //both lists are empty
+            secondList.clear();
+            newList = list.concat(secondList);
+            expect(newList.getSize(), 0);
+            expect(newList.getHeadNode()).to.not.exist;
+            expect(newList.getTailNode()).to.not.exist;
```
Comment (jasonsjones) [inline test/list.spec.js:285]: This can be another separate tests. Maybe...

```
it('concats two empty lists', () => {
    // set up and expectations
});
```

File: test/list.spec.js [hunk 3]
```diff
@@ -265,6 +265,27 @@ describe('Linked List', function() {
...
+
+            //first list is populated, second is empty
+            populateList(list, 3);
+            newList = list.concat(secondList);
+            expect(newList.getSize(), 3);
+            expect(newList.getHeadNode()).to.not.equal(null);
+            expect(newList.getTailNode()).to.not.equal(null);
```
Comment (jasonsjones) [inline test/list.spec.js:294]: And another test...  I hope you get the idea.

**Review by jasonsjones — CHANGES_REQUESTED** [review]
I really need to check my notification settings 😉, I am just now seeing this.  Thanks for taking the time to follow up on your PR to fix this issue.

Made a few suggestions that should be pretty easy to update.  Let me know if you can split up the tests...if not, no worries, I can get this in and update at a later date.  Thanks again!

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "dbly-linked-list",
-    "version": "0.3.0",
+    "version": "0.3.1",
```
Comment (jasonsjones) [inline package.json:3]: Thanks for the version bump 👍

**Comment by bmcgary** [comment]
Just pushed a fix. Should be good to good now! 👍

**Review by jasonsjones — APPROVED** [review]

**Comment by jasonsjones** [comment]
Awesome!  Thanks, Blaine!
